src/main/java/com/icthh/xm/ms/entity/domain/FunctionContext.java

```diff
@@ -246,12 +257,24 @@ public boolean equals(Object o) {
     }
 
     public Object functionResult() {
+        if (getBinaryData().isPresent()) {
+            return getBinaryData().get();
+        }
+
         if (onlyData) {
             return data.get("data");
         }
         return this;
     }
 
+    public boolean isBinaryData() {
+        return binaryDataField != null;
+    }
+
+    public Optional<Object> getBinaryData() {
```

Comment:
Do we really need to return Optional here?

Reply:
indeed, Optional can be avoided, removed it